statsd/src/main/scala/zio/metrics/connectors/statsd/package.scala

```diff
@@ -5,11 +5,28 @@ import zio.metrics.connectors.internal.MetricsClient
 
 package object statsd {
 
+  @deprecated("Use zio.metrics.connectors.datadog.live instead", "2.4.0")
   lazy val statsdLayer: ZLayer[StatsdConfig & MetricsConfig, Nothing, Unit] =
     ZLayer.scoped(
       StatsdClient.make.flatMap(clt => MetricsClient.make(statsdHandler(clt))).unit,
     )
 
+  lazy val live: ZLayer[StatsdConfig & MetricsConfig, Nothing, StatsdClient] =
```

Comment:
I think we. can be a bit smarter with the names here. One can be `statsdUDP` and another `statsdUDS` or something alone these lines.

We should also add some short scaladoc explaining their difference